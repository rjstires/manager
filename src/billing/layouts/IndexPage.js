import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import { push } from 'react-router-redux';

import { Tabs } from 'linode-components/tabs';

import { setAnalytics, setTitle } from '~/actions';
import { account, invoices } from '~/api';


export class IndexPage extends Component {
  static async preload({ dispatch }) {
    await Promise.all([account.one(), invoices.all()].map(r => dispatch(r)));
  }

  componentDidMount() {
    const { dispatch } = this.props;
    dispatch(setTitle('Billing'));
    dispatch(setAnalytics('billing'));
  }

  render() {
    const tabs = [
      { name: 'Dashboard', link: '' },
      { name: 'History', link: 'history' },
    ].map(t => ({ ...t, link: `/billing/${t.link}` }));

    return (
      <div>
        <header className="main-header">
          <div className="container">
            <h1 className="float-sm-left">Billing</h1>
          </div>
        </header>
        <div className="main-header-fix"></div>
        <Tabs
          tabs={tabs}
          onClick={(e, tabIndex) => {
            e.stopPropagation();
            this.props.dispatch(push(tabs[tabIndex].link));
          }}
          pathname={location.pathname}
        >
          {this.props.children}
        </Tabs>
      </div>
    );
  }
}

IndexPage.propTypes = {
  dispatch: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
};

export default connect()(IndexPage);
