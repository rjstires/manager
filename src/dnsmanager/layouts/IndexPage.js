import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router';
import moment from 'moment';
import _ from 'lodash';

import { setError } from '~/actions/errors';
import { showModal, hideModal } from '~/actions/modal';
import { List, Table } from '~/components/tables';
import { MassEditControl } from '~/components/tables/controls';
import { ListHeader } from '~/components/tables/headers';
import { ListBody, ListGroup } from '~/components/tables/bodies';
import DeleteModalBody from '~/components/modals/DeleteModalBody';
import {
  ButtonCell,
  CheckboxCell,
  LinkCell,
} from '~/components/tables/cells';
import { dnszones as apiDnszones } from '~/api';
import { setSource } from '~/actions/source';
import { setTitle } from '~/actions/title';
import { toggleSelected, toggleSelectAll } from '../actions';
import CreateHelper from '~/components/CreateHelper';

export class IndexPage extends Component {
  static async preload({ dispatch }) {
    try {
      await dispatch(apiDnszones.all());
    } catch (response) {
      // eslint-disable-next-line no-console
      console.error(response);
      dispatch(setError(response));
    }
  }

  constructor() {
    super();
    this.deleteZone = this.deleteZone.bind(this);
    this.remove = this.remove.bind(this);
  }

  async componentDidMount() {
    const { dispatch } = this.props;
    dispatch(setSource(__filename));

    dispatch(setTitle('DNS Manager'));
  }

  remove(zones) {
    const { dispatch, selected, dnszones } = this.props;
    dispatch(showModal('Confirm deletion',
      <DeleteModalBody
        buttonText="Delete selected zones"
        onOk={() => {
          const zoneIds = zones.map((zone) => zone.id);

          zoneIds.forEach(function (id) {
            dispatch(apiDnszones.delete(id));
          });
          dispatch(toggleSelected(zoneIds));
          dispatch(hideModal());
        }}
        items={dnszones.dnszones}
        selectedItems={Object.keys(selected)}
        typeOfItem="zones"
        label="dnszone"
        onCancel={() => dispatch(hideModal())}
      />
    ));
  }

  deleteZone(zoneId) {
    const { dispatch } = this.props;
    dispatch(showModal('Delete DNS Zone', this.renderModal(zoneId)));
  }

  renderModal(zoneId) {
    const { dispatch, dnszones: theseZones } = this.props;
    return (
      <DeleteModalBody
        buttonText="Delete selected zones"
        onOk={async () => {
          await dispatch(apiDnszones.delete(zoneId));
          dispatch(toggleSelectAll());
          dispatch(hideModal());
        }}
        items={theseZones.dnszones}
        selectedItems={zoneId}
        typeOfItem="zones"
        label="dnszone"
        onCancel={() => dispatch(hideModal())}
      />
    );
  }

  renderZones(zones) {
    const { dispatch, selected } = this.props;
    // TODO: add sort function in dns zones config definition
    const sortedZones = _.sortBy(Object.values(zones), ({ created }) => moment(created));

    const groups = _.sortBy(
      _.map(_.groupBy(sortedZones, ({ display_group: group }) => group), (_zones, _group) => {
        return {
          name: _group,
          data: _zones,
        };
      }), zoneGroup => zoneGroup.group);

    return (
      <List>
        <ListHeader>
          <div className="pull-sm-left">
            <MassEditControl
              data={sortedZones}
              dispatch={dispatch}
              massEditOptions={[
                { name: 'Delete', action: this.remove },
              ]}
              selectedMap={selected}
              toggleSelected={toggleSelected}
            />
          </div>
        </ListHeader>
        <ListBody>
          {groups.map((group, index) => {
            return (
              <ListGroup
                key={index}
                name={group.name}
              >
                <Table
                  columns={[
                    { cellComponent: CheckboxCell },
                    {
                      className: 'RowLabelCell',
                      cellComponent: LinkCell,
                      hrefFn: (zone) => `/dnsmanager/${zone.dnszone}`, textKey: 'dnszone',
                    },
                    { dataKey: 'type' },
                    {
                      cellComponent: ButtonCell,
                      text: 'Delete',
                      onClick: (zone) => { this.deleteZone(zone.id); },
                    },
                  ]}
                  data={group.data}
                  selectedMap={selected}
                  disableHeader
                  onToggleSelect={(record) => {
                    dispatch(toggleSelected(record.id));
                  }}
                />
              </ListGroup>
            );
          })}
        </ListBody>
      </List>
    );
  }

  render() {
    const { dnszones, selected } = this.props;

    return (
      <div className="PrimaryPage container">
        <header className="PrimaryPage-header">
          <div className="PrimaryPage-headerRow clearfix">
            <h1 className="float-sm-left">DNS Manager</h1>
            <Link to="/dnsmanager/create" className="linode-add btn btn-primary float-sm-right">
              <span className="fa fa-plus"></span>
              Add a DNS Zone
            </Link>
          </div>
        </header>
        <div className="PrimaryPage-body">
          {Object.keys(dnszones.dnszones).length ? this.renderZones(dnszones.dnszones, selected) :
            <CreateHelper label="zones" href="/dnsmanager/create" linkText="Add a zone" />}
        </div>
      </div>
    );
  }
}

IndexPage.propTypes = {
  dispatch: PropTypes.func,
  dnszones: PropTypes.object,
  selected: PropTypes.object,
};


function select(state) {
  return {
    dnszones: state.api.dnszones,
    selected: state.dnsmanager.index.selected,
  };
}

export default connect(select)(IndexPage);
