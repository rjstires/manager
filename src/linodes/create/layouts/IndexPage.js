import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import { push } from 'react-router-redux';

import SourceSelection from '../components/SourceSelection';
import ServiceSelection from '../components/ServiceSelection';
import DatacenterSelection from '../components/DatacenterSelection';
import Details from '../components/Details';
import { fetchAllDistros } from '~/actions/api/distros';
import { fetchAllDatacenters } from '~/actions/api/datacenters';
import { fetchAllServices } from '~/actions/api/services';
import { fetchLinodes, createLinode } from '~/actions/api/linodes';
import { fetchAllBackups } from '~/actions/api/backups';
import { setError } from '~/actions/errors';

export class IndexPage extends Component {
  constructor() {
    super();
    this.onSubmit = this.onSubmit.bind(this);
    this.createLinode = this.createLinode.bind(this);
    this.renderProgress = this.renderProgress.bind(this);
    this.state = {
      service: null,
      datacenter: null,
      distribution: null,
      backup: null,
      sourceTab: 0,
      errors: {},
      loading: false,
      progress: -1,
    };
  }

  async componentDidMount() {
    const { dispatch } = this.props;
    try {
      await Promise.all([
        dispatch(fetchAllDistros()),
        dispatch(fetchAllDatacenters()),
        dispatch(fetchAllServices()),
        dispatch(fetchLinodes()),
      ]);
    } catch (response) {
      dispatch(setError(response));
    }
    const { location } = this.props;
    if (location.query && location.query.linode && location.query.backup) {
      let { linodes } = this.props;
      let linode = linodes.linodes[location.query.linode];
      if (linode) {
        await dispatch(fetchAllBackups(location.query.linode, location.query.backup));
        linodes = this.props.linodes;
        linode = linodes.linodes[location.query.linode];
        const backup = linode._backups.backups[location.query.backup];
        if (backup) {
          this.setState({
            backup: backup.id,
            datacenter: backup.datacenter.id,
            sourceTab: 1,
          });
        }
      }
    }
  }

  async onSubmit({ group, labels, password, backups }) {
    const { dispatch } = this.props;
    try {
      if (labels.length === 1) {
        this.setState({ loading: true });
        const [label] = labels;
        const linode = await this.createLinode({ group, label, password, backups });
        this.setState({ loading: false });
        dispatch(push(`/linodes/${linode.id}`));
      } else {
        this.setState({ loading: true, progress: 0 });
        for (let i = 0; i < labels.length; ++i) {
          const label = labels[i] || `${labels[0]}-${i}`;
          await this.createLinode({ group, label, password, backups });
          this.setState({ progress: (i + 1) / labels.length });
        }
        this.setState({ loading: false });
        dispatch(push('/linodes'));
      }
    } catch (response) {
      const { errors } = await response.json();
      const errorsByField = {};
      errors.forEach(({ field, reason }) => {
        if (!(field in errorsByField)) errorsByField[field] = [];
        errorsByField[field].push(reason);
      });
      this.setState({ progress: -1, loading: false, errors: errorsByField });
    }
  }

  async createLinode({ group, label, password, backups }) {
    const { dispatch } = this.props;
    const { service, datacenter, distribution, backup } = this.state;
    return await dispatch(createLinode({
      root_pass: password,
      service,
      distribution,
      backup,
      datacenter,
      label,
      group,
      backups,
    }));
  }

  renderProgress() {
    const { progress } = this.state;
    return (
      <div className="create-page">
        <h1>Add a Linode</h1>
        <div className="card page-card">
          <header>
            <h2>Provisioning Linodes</h2>
          </header>
          <div className="card-body">
            <p>Your Linodes are being provisioned.</p>
            <progress
              className="progress"
              value={progress}
              max="1"
            ></progress>
          </div>
        </div>
      </div>
    );
  }

  render() {
    const {
      distros,
      linodes,
      datacenters,
      services,
    } = this.props;
    const {
      backup,
      distribution,
      datacenter,
      service,
      sourceTab,
      loading,
      progress,
    } = this.state;

    if (progress !== -1) {
      return this.renderProgress();
    }

    return (
      <div className="create-page">
        <h1>Add a Linode</h1>
        <div className="card page-card">
          <SourceSelection
            distribution={distribution}
            backup={backup}
            selectedTab={sourceTab}
            distros={distros.distributions}
            onTabChange={ix => this.setState({ sourceTab: ix })}
            onSourceSelected={(type, id, linodeId) => {
              if (type === 'backup' && linodeId && id) {
                const linode = linodes.linodes[linodeId];
                const backup = linode._backups.backups[id];
                this.setState({
                  backup: id,
                  datacenter: backup.datacenter.id,
                  distribution: null,
                });
              } else {
                this.setState({ [type]: id, backup: null });
              }
            }}
            linodes={linodes}
          />
        </div>
        <div className="card page-card">
          <DatacenterSelection
            selected={datacenter}
            datacenters={datacenters.datacenters}
            disabled={backup !== null}
            onDatacenterSelected={id => this.setState({ datacenter: id })}
          />
        </div>
        <div className="card page-card">
          <ServiceSelection
            selected={service}
            services={services.services}
            onServiceSelected={id => this.setState({ service: id })}
          />
        </div>
        <div className="card page-card">
          <Details
            onSubmit={this.onSubmit}
            submitEnabled={!!(distribution || backup) && !!datacenter && !!service && !loading}
            errors={this.state.errors}
          />
        </div>
      </div>
    );
  }
}

IndexPage.propTypes = {
  dispatch: PropTypes.func.isRequired,
  distros: PropTypes.object,
  linodes: PropTypes.object,
  services: PropTypes.object,
  datacenters: PropTypes.object,
  location: PropTypes.object,
};

function select(state) {
  return {
    distros: state.api.distributions,
    linodes: state.api.linodes,
    datacenters: state.api.datacenters,
    services: state.api.services,
  };
}

export default connect(select)(IndexPage);
