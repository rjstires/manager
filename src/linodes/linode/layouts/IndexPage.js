import React, { Component, PropTypes } from 'react';
import { Tab, Tabs, TabList, TabPanel } from 'react-tabs';
import { connect } from 'react-redux';
import { Link } from 'react-router';
import { push } from 'react-router-redux';

import Dropdown from '~/components/Dropdown';
import { LinodeStates, LinodeStatesReadable } from '~/constants';
import { setError } from '~/actions/errors';
import {
  fetchLinode,
  fetchAllLinodeConfigs,
  powerOnLinode,
  powerOffLinode,
  rebootLinode,
} from '~/actions/api/linodes';

export function getLinode() {
  const { linodes } = this.props.linodes;
  const linodeId = parseInt(this.props.params.linodeId);
  return linodes ? linodes[linodeId] : null;
}

export async function loadLinode() {
  const { dispatch } = this.props;
  let linode = this.getLinode();
  if (!linode) {
    const linodeId = parseInt(this.props.params.linodeId);
    try {
      await dispatch(fetchLinode(linodeId));
      linode = this.getLinode();
    } catch (response) {
      dispatch(setError(response));
    }
  }
  if (linode && (!linode._configs || linode._configs.totalPages === -1)) {
    await dispatch(fetchAllLinodeConfigs(linode.id));
  }
}

export function renderTabs(tabList, selected, isSubtab = false) {
  const { dispatch } = this.props;

  return (
    <Tabs
      onSelect={ix => dispatch(push(tabList[ix].link))}
      selectedIndex={tabList.indexOf(selected)}
    >
      <TabList>
        {tabList.map(t => (
          <Tab key={t.name}>
            <Link to={t.link} onClick={e => e.preventDefault()}>{t.name}</Link>
          </Tab>
        ))}
      </TabList>
      {tabList.map(t => (
        <TabPanel key={t.name}>
          <div className={isSubtab ? '' : 'container'}>
            {t === selected ? this.props.children : null}
          </div>
        </TabPanel>
       ))}
    </Tabs>
  );
}

export class IndexPage extends Component {
  constructor() {
    super();
    this.getLinode = getLinode.bind(this);
    this.render = this.render.bind(this);
    this.renderLabel = this.renderLabel.bind(this);
    this.renderTabs = renderTabs.bind(this);
    this.loadLinode = module.exports.loadLinode.bind(this);
    this.componentDidMount = this.componentDidMount.bind(this);
    this.state = { config: '', label: '', group: '' };
  }

  async componentDidMount() {
    await this.loadLinode();
    const linode = this.getLinode();
    const defaultConfig = Object.values(linode._configs.configs)[0];
    this.setState({
      config: defaultConfig ? defaultConfig.id : '',
      label: linode.label,
      group: linode.group,
    });
  }

  renderLabel(linode) {
    const label = linode.group ?
      <span>{linode.group} / {linode.label}</span> :
      <span>{linode.label}</span>;

    return (
      <div style={{ display: 'inline-block' }}>
        <h1>{label}</h1>
      </div>
    );
  }

  renderHeader() {
    const linode = this.getLinode();
    const { dispatch } = this.props;

    const dropdownElements = [
      {
        name: <span><i className="fa fa-refresh"></i> Reboot</span>,
        _action: rebootLinode,
        _condition: () => linode.state !== 'offline',
      },
      {
        name: <span><i className="fa fa-power-off"></i> Power Off</span>,
        _action: powerOffLinode,
        _condition: () => linode.state === 'running',
      },
      {
        name: <span><i className="fa fa-power-off"></i> Power On</span>,
        _action: powerOnLinode,
        _condition: () => linode.state === 'offline',
      },
    ]
    .filter(element => element._condition())
    .map(element => ({
      ...element,
      action: () => dispatch(element._action(linode.id, this.state.config || null)),
    }));

    const renderConfigSelect = linode._configs.totalResults > 1 &&
      LinodeStates.pending.indexOf(linode.state) === -1;

    return (
      <header className="main-header">
        <div className="container">
          {this.renderLabel(linode)}
          {LinodeStates.pending.indexOf(linode.state) !== -1 ? null :
            <span className="pull-right">
              <Dropdown elements={dropdownElements} leftOriented={false} />
            </span>}
          {!renderConfigSelect ? null :
            <span
              className="pull-right configs"
              style={{ paddingRight: '15px', lineHeight: '30px' }}
            >
              <select
                className="form-control"
                value={this.state.config}
                onChange={e => this.setState({ config: e.target.value })}
              >
                {Object.values(linode._configs.configs).map(config =>
                  <option key={config.id} value={config.id}>{config.label}</option>)}
              </select>
            </span>}
          <span
            className={`pull-right linode-status ${linode.state}`}
            style={{ paddingRight: '15px', lineHeight: '30px' }}
          >
            {LinodeStatesReadable[linode.state]}
          </span>
        </div>
      </header>
    );
  }

  render() {
    const linode = this.getLinode();
    if (!linode) return <span></span>;

    const tabList = [
      { name: 'Dashboard', link: '' },
      { name: 'Networking', link: '/networking' },
      { name: 'Rebuild', link: '/rebuild' },
      { name: 'Resize', link: '/resize' },
      { name: 'Rescue', link: '/rescue' },
      { name: 'Backups', link: '/backups' },
      { name: 'Settings', link: '/settings' },
    ].map(t => ({ ...t, link: `/linodes/${linode.id}${t.link}` }));

    const pathname = location ? location.pathname : tabList[0].link;
    const selected = tabList.reduce((last, current) =>
      (pathname.indexOf(current.link) === 0 ? current : last));

    return (
      <div className="details-page">
        {this.renderHeader(linode)}
        <div className="main-header-fix"></div>
        {this.renderTabs(tabList, selected)}
      </div>
    );
  }
}

IndexPage.propTypes = {
  dispatch: PropTypes.func,
  username: PropTypes.string,
  linodes: PropTypes.object,
  params: PropTypes.shape({
    linodeId: PropTypes.string,
  }),
  detail: PropTypes.object,
  children: PropTypes.node,
  location: PropTypes.object,
  router: PropTypes.object,
  route: PropTypes.object,
};

function select(state) {
  return { linodes: state.api.linodes };
}

export default connect(select)(IndexPage);
