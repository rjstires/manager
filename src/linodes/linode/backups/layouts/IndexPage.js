import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import { getLinode, loadLinode, renderTabs } from '~/linodes/linode/layouts/IndexPage';
import { parallel } from '~/api/util';
import { linodes } from '~/api';
import { enableBackup } from '~/api/backups';
import { setSource } from '~/actions/source';

export class IndexPage extends Component {
  constructor() {
    super();
    this.getLinode = getLinode.bind(this);
    this.enableBackups = this.enableBackups.bind(this);
    this.componentDidMount = this.componentDidMount.bind(this);
    this.renderTabs = renderTabs.bind(this);
    this.loadLinode = loadLinode.bind(this);
  }

  async componentDidMount() {
    const { dispatch } = this.props;
    dispatch(setSource(__filename));
    const { linodeId } = this.props.params;
    await dispatch(parallel(
      linodes.backups.all(linodeId),
      linodes.all()));
  }

  enableBackups() {
    const { dispatch } = this.props;
    dispatch(enableBackup(this.getLinode().id));
  }

  render() {
    const linode = this.getLinode();
    if (!linode) return null;

    if (!linode.backups.enabled) {
      return (
        <section className="card">
          <form onSubmit={this.enableBackups}>
            <p>
              Backups not enabled. Enable backups for
              ${(linode.type[0].backups_price / 100).toFixed(2)}.
            </p>
            <button className="btn btn-primary">Enable backups</button>
          </form>
        </section>
      );
    }

    const tabList = [
      { name: 'Summary', link: '/' },
      { name: 'History', link: '/history' },
      { name: 'Settings', link: '/settings' },
    ].map(t => ({ ...t, link: `/linodes/${linode.id}/backups${t.link}` }));

    const pathname = location ? location.pathname : tabList[0].link;
    const selected = tabList.reduce((last, current) =>
      (pathname.indexOf(current.link) === 0 ? current : last));

    return this.renderTabs(tabList, selected, true);
  }
}

IndexPage.propTypes = {
  dispatch: PropTypes.func.isRequired,
  linodes: PropTypes.object.isRequired,
  params: PropTypes.shape({
    linodeId: PropTypes.string.isRequired,
  }).isRequired,
  children: PropTypes.object,
  location: PropTypes.object,
  router: PropTypes.object,
  route: PropTypes.object,
};

function select(state) {
  return {
    linodes: state.api.linodes,
  };
}

export default withRouter(connect(select)(IndexPage));
