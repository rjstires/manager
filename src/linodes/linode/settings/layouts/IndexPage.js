import { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';

import { getLinode, loadLinode, renderTabs } from '~/linodes/linode/layouts/IndexPage';

export class IndexPage extends Component {
  constructor() {
    super();
    this.getLinode = getLinode.bind(this);
    this.renderTabs = renderTabs.bind(this);
    this.loadLinode = loadLinode.bind(this);
  }

  componentDidMount() {
    this.loadLinode();
  }

  render() {
    const linode = this.getLinode();
    if (!linode) return null;

    const tabList = [
      { name: 'Display', link: '/' },
      { name: 'Alerts', link: '/alerts' },
      { name: 'Advanced', link: '/advanced' },
    ].map(t => ({ ...t, link: `/linodes/${linode.id}/settings${t.link}` }));

    const pathname = location ? location.pathname : tabList[0].link;
    const selected = tabList.reduce((last, current) =>
      (pathname.indexOf(current.link) === 0 ? current : last));

    return this.renderTabs(tabList, selected, true);
  }
}

IndexPage.propTypes = {
  dispatch: PropTypes.func.isRequired,
  linodes: PropTypes.object.isRequired,
  params: PropTypes.shape({
    linodeId: PropTypes.string.isRequired,
  }).isRequired,
  children: PropTypes.object,
  location: PropTypes.object,
  router: PropTypes.object,
  route: PropTypes.object,
};

function select(state) {
  return {
    linodes: state.api.linodes,
  };
}

export default withRouter(connect(select)(IndexPage));
