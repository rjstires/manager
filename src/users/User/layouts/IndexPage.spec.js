import React from 'react';
import { shallow } from 'enzyme';
import { IndexPage } from './IndexPage';

describe('users/IndexPage', () => {
  it('should render without error', () => {
    const mockDispatch = jest.fn();
    const wrapper = shallow(
      <IndexPage dispatch={mockDispatch} user={{}} children={<div></div>} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
