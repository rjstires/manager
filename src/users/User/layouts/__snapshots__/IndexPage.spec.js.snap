// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`users/IndexPage should render without error 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IndexPage
    dispatch={[Function]}
    user={Object {}}
  >
    <div />
  </IndexPage>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <SideEffect(ChainedDocumentTitle)
          title={undefined}
        />,
        <header
          className="main-header"
        >
          <div
            className="container"
          >
            <div
              className="float-sm-left"
            >
              <Link
                onlyActiveOnIndex={false}
                style={Object {}}
                to="/users"
              >
                Users
              </Link>
              <h1
                title={undefined}
              >
                <Link
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="/users/undefined"
                />
              </h1>
            </div>
          </div>
        </header>,
        <div
          className="main-header-fix"
        />,
        <Tabs
          className=""
          isSubTabs={false}
          onClick={[Function]}
          pathname="blank"
          tabs={
            Array [
              Object {
                "link": "/users/undefined",
                "name": "Dashboard",
              },
            ]
          }
        >
          <div />
        </Tabs>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "title": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="container"
          >
            <div
              className="float-sm-left"
            >
              <Link
                onlyActiveOnIndex={false}
                style={Object {}}
                to="/users"
              >
                Users
              </Link>
              <h1
                title={undefined}
              >
                <Link
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="/users/undefined"
                />
              </h1>
            </div>
          </div>,
          "className": "main-header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="float-sm-left"
            >
              <Link
                onlyActiveOnIndex={false}
                style={Object {}}
                to="/users"
              >
                Users
              </Link>
              <h1
                title={undefined}
              >
                <Link
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="/users/undefined"
                />
              </h1>
            </div>,
            "className": "container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Link
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="/users"
                >
                  Users
                </Link>,
                <h1
                  title={undefined}
                >
                  <Link
                    onlyActiveOnIndex={false}
                    style={Object {}}
                    to="/users/undefined"
                  />
                </h1>,
              ],
              "className": "float-sm-left",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Users",
                  "onlyActiveOnIndex": false,
                  "style": Object {},
                  "to": "/users",
                },
                "ref": null,
                "rendered": "Users",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    onlyActiveOnIndex={false}
                    style={Object {}}
                    to="/users/undefined"
                  />,
                  "title": undefined,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": undefined,
                    "onlyActiveOnIndex": false,
                    "style": Object {},
                    "to": "/users/undefined",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "h1",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "main-header-fix",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <div />,
          "className": "",
          "isSubTabs": false,
          "onClick": [Function],
          "pathname": "blank",
          "tabs": Array [
            Object {
              "link": "/users/undefined",
              "name": "Dashboard",
            },
          ],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SideEffect(ChainedDocumentTitle)
            title={undefined}
          />,
          <header
            className="main-header"
          >
            <div
              className="container"
            >
              <div
                className="float-sm-left"
              >
                <Link
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="/users"
                >
                  Users
                </Link>
                <h1
                  title={undefined}
                >
                  <Link
                    onlyActiveOnIndex={false}
                    style={Object {}}
                    to="/users/undefined"
                  />
                </h1>
              </div>
            </div>
          </header>,
          <div
            className="main-header-fix"
          />,
          <Tabs
            className=""
            isSubTabs={false}
            onClick={[Function]}
            pathname="blank"
            tabs={
              Array [
                Object {
                  "link": "/users/undefined",
                  "name": "Dashboard",
                },
              ]
            }
          >
            <div />
          </Tabs>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "title": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="container"
            >
              <div
                className="float-sm-left"
              >
                <Link
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="/users"
                >
                  Users
                </Link>
                <h1
                  title={undefined}
                >
                  <Link
                    onlyActiveOnIndex={false}
                    style={Object {}}
                    to="/users/undefined"
                  />
                </h1>
              </div>
            </div>,
            "className": "main-header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="float-sm-left"
              >
                <Link
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="/users"
                >
                  Users
                </Link>
                <h1
                  title={undefined}
                >
                  <Link
                    onlyActiveOnIndex={false}
                    style={Object {}}
                    to="/users/undefined"
                  />
                </h1>
              </div>,
              "className": "container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Link
                    onlyActiveOnIndex={false}
                    style={Object {}}
                    to="/users"
                  >
                    Users
                  </Link>,
                  <h1
                    title={undefined}
                  >
                    <Link
                      onlyActiveOnIndex={false}
                      style={Object {}}
                      to="/users/undefined"
                    />
                  </h1>,
                ],
                "className": "float-sm-left",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Users",
                    "onlyActiveOnIndex": false,
                    "style": Object {},
                    "to": "/users",
                  },
                  "ref": null,
                  "rendered": "Users",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      onlyActiveOnIndex={false}
                      style={Object {}}
                      to="/users/undefined"
                    />,
                    "title": undefined,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": undefined,
                      "onlyActiveOnIndex": false,
                      "style": Object {},
                      "to": "/users/undefined",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "h1",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "main-header-fix",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <div />,
            "className": "",
            "isSubTabs": false,
            "onClick": [Function],
            "pathname": "blank",
            "tabs": Array [
              Object {
                "link": "/users/undefined",
                "name": "Dashboard",
              },
            ],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
